features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/JsonPathHelper.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.config.rest.impl;
+
+import java.util.List;
+import java.util.Random;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+
+public class JsonPathHelper {
+
+    private static final Configuration AS_PATH_LIST = Configuration.builder()
+            .options(Option.AS_PATH_LIST).build();
+    
+    public static String get(String data, String path) {
+        JsonPath jsonPath = JsonPath.compile(path);
+
+        //it's ok to read multiple nodes, it could be useful 
+        //assertThereIsExactlyOnePath(data, jsonPath);
+        
+        return JsonPath.parse(data).read(jsonPath).toString();
+    }
+
+    public static String update(String data, String path, String newPartContent) {
+        JsonPath jsonPath = JsonPath.compile(path);
+        assertThereIsExactlyOnePath(data, jsonPath);
+        
+        //the object under specified path will be replaced with providet text without checking if the text is valid JSON 
+        //for that on the first step the specified node will be replaced by unique number, and then this number will be replaced 
+        //withh provided text
+        Long unique = getUnique(data);
+        String newJson = JsonPath.parse(data).set(jsonPath, unique).jsonString();
```

Comment:
@freddy61025 I think the idea here is: parse the data into json, resolve the path, insert a marker into the according position, deserialize again and replace the marker in the deserialized text with the value we want to insert.

@dherdt If the new content is no valid json, we should not insert it here. This would lead to invalid json in the resulting value. Instead, I think, we should parse the new value into json and insert it into the data in the parsed form.

Reply:
@freddy61025
Unique number is used because JsonPath would replace complex structure with a string.

@fooker 
Freddy said I should make no validation because the CM will make it.

@fooker & @freddy61025 
Please let me know if I need to check if it is a valid JSON (I think there is no need for the check)

Reply:
When save CM will do generic validation + validation callbacks. It is up to you do extra validation or not. I would suggest no need to have extra validation. If any case it fail to check, we should fix the validation in CM.

Reply:
For my question. I want to understand what is the difference between. Is that because using JsonPath set will escape the newPartContent? If it must be in that way you better add a comment about why do in 2 steps.

`
        String newJson = JsonPath.parse(data).set(jsonPath, unique).jsonString();
        return newJson.replace(unique.toString(), newPartContent);
`

And
`
        return JsonPath.parse(data).set(jsonPath, newPartContent).jsonString();
`

Reply:
I think there is no need to verify. But to update the object, we need to pares it to json first. `newPartContent` must be structured in some way to make this work.